integration/graphql-type-factories.ts

```diff
@@ -129,19 +130,38 @@ factories["CalendarInterval"] = newCalendarInterval;
 
 export interface ChildOptions {
   __typename?: "Child";
-  parent?: Child["parent"];
+  name?: Child["name"];
+  parent?: NamedOptions;
 }
 
 export function newChild(options: ChildOptions = {}, cache: Record<string, any> = {}): Child {
   const o = (options.__typename ? options : cache["Child"] = {}) as Child;
   (cache.all ??= new Set()).add(o);
   o.__typename = "Child";
+  o.name = options.name ?? "name";
   o.parent = maybeNew("Author", options.parent, cache, options.hasOwnProperty("parent"));
   return o;
 }
 
 factories["Child"] = newChild;
 
+export interface ParentOptions {
+  __typename?: "Parent";
+  children?: Array<NamedOptions>;
+  name?: Parent["name"];
+}
+
+export function newParent(options: ParentOptions = {}, cache: Record<string, any> = {}): Parent {
+  const o = (options.__typename ? options : cache["Parent"] = {}) as Parent;
+  (cache.all ??= new Set()).add(o);
+  o.__typename = "Parent";
+  o.children = (options.children ?? []).map((i) => maybeNew("Named", i, cache, options.hasOwnProperty("children")));
```

Comment:
NOTE: This is one of the main changes - before would have been:
```suggestion
  o.children = options.children ?? [];
```